### PR TITLE
jsonize toolweapon actions

### PR DESCRIPTION
**Pull request by nornagon**
#### Summary
Infrastructure "Added transform_item and turn_cost EOC effects, and has_ammo condition"

#### Purpose of change

Expands the capabilities of item EOCs enough to JSONize all the
toolweapon effects (trimmer, carver, chainsaw, masonrysaw etc.).

#### Describe the solution

* Replaced the toolweapon use_actions and tick_actions by EOCs.
* Added new `has_ammo` conditional to allow the EOC to fail if the chainsaw is
  out of ammo. This calls `ammo_sufficient` under the hood.
* Added new `transform_item` effect which calls `convert`.
* Added new `turn_cost` effect to add a move cost to an activation EOC. Counts
  in turns instead of moves because this way I could reuse the duration_var
  infrastructure, so you can specify this in seconds in a string if you want.

The only thing I haven't ported over is the chainsaw idle / theme ambient sounds. I'll add an EOC effect to control ambient sounds once #70898 lands.

#### Describe alternatives you've considered

Maybe this is better off staying in C++? I feel like the JSON is more flexible
and moddable, and it seems weird to me for chainsaws to be "blessed" in this
way. But the new JSON is quite a bit more verbose than the old JSON. It would
be nice if there were a less-verbose way to say "invoke this EOC with these
variables".

#### Testing

* Activate a charged combat chainsaw, then checked:
  * Move cost was applied
  * Message was printed (NB this is a bugfix! In current experimental, no
    message is printed when you successfully start a chainsaw.)
  * The item was converted to it's \_on variant
  * The running sounds messages are printed every now and then
* Manually deactivate a charged combat chainsaw (on)
  * The deactivate message is printed
  * The item is transformed to \_off
* Activate a charged combat chainsaw while underwater
  * The failure message is printed
  * The item is not transformed
* Activate a combat chainsaw with no fuel
  * The failure message is printed
  * The item is not transformed
* Activate a charged combat chainsaw, then move underwater
  * The drown message is printed
  * The item is reverted to \_off

### CODE REVIEW

**Comment by GuardianDll** [comment]
Isn't there only a chance to start a saw when you activate it? Or it's only for generic chainsaw?

**Comment by nornagon** [comment]
> Isn't there only a chance to start a saw when you activate it? Or it's only for generic chainsaw?

Yeah, that's only for the generic chainsaw, the combat saws are guaranteed-start. Though with this change it'd be possible to tweak that in JSON :)

**Comment by nornagon** [comment]
Also jsonized carver_off here. It had a "works underwater" flag set in the C++, but the JSON for that object has the "WATER_BREAK_ACTIVE" flag, which makes more sense.

File: data/json/items/melee/swords_and_blades.json [hunk 1]
```diff
@@ -2867,7 +2867,25 @@
     "ammo": [ "gasoline" ],
     "charges_per_use": 5,
     "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "watertight": true, "ammo_restriction": { "gasoline": 450 } } ],
-    "use_action": [ "COMBATSAW_OFF" ],
+    "use_action": {
+      "type": "effect_on_conditions",
+      "menu_text": "Turn on",
+      "effect_on_conditions": [
+        {
+          "id": "EOC_toolweapon_activate__combatsaw",
```
Comment (GuardianDll) [inline data/json/items/melee/swords_and_blades.json:2875]: why all ids have double underscore?

**Comment by nornagon** [comment]
I did that to separate the name of the EOC from the name of the variant.
It’s a bit weird to me that these inline EOCs need an ID.


j

**Comment by Maleclypse** [comment]
I'm going to let you handle the conflict on this one.

**Comment by GuardianDll** [comment]
May i ask to document new effects that was added in this PR? specifically `turn_cost`, `has_ammo` and `transform_item` in [EFFECT_ON_CONDITION.md](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/EFFECT_ON_CONDITION.md#effects), and `damage_level` in [NPCs.md](https://github.com/CleverRaven/Cataclysm-DDA/blob/master/doc/NPCs.md#dialogue-functions) (unless i miss some more changes)

**Pull request merged**